### PR TITLE
Add configuration for expiring minion jobs

### PR DESCRIPTION
**Pull request by perlpunk**
Issue: https://progress.opensuse.org/issues/111152

### CODE REVIEW

File: lib/OpenQA/Setup.pm [hunk 1]
```diff
@@ -148,6 +149,7 @@ sub read_config ($app) {
             screenshot_cleanup_batch_size => OpenQA::Task::Job::Limit::DEFAULT_SCREENSHOTS_PER_BATCH,
             screenshot_cleanup_batches_per_minion_job => OpenQA::Task::Job::Limit::DEFAULT_BATCHES_PER_MINION_JOB,
             results_min_free_disk_space_percentage => undef,
+            expire_minion_jobs => ONE_DAY * 7,
```
Comment (okurz) [inline lib/OpenQA/Setup.pm:152]:
```suggestion
            expire_minion_jobs => ONE_WEEK,
```

File: t/config.t [hunk 2]
```diff
@@ -131,6 +132,7 @@ subtest 'Test configuration default modes' => sub {
             asset_cleanup_max_free_percentage => 100,
             screenshot_cleanup_batch_size => OpenQA::Task::Job::Limit::DEFAULT_SCREENSHOTS_PER_BATCH,
             screenshot_cleanup_batches_per_minion_job => OpenQA::Task::Job::Limit::DEFAULT_BATCHES_PER_MINION_JOB,
+            expire_minion_jobs => ONE_DAY * 7,
```
Comment (okurz) [inline t/config.t:135]:
```suggestion
            expire_minion_jobs => ONE_WEEK,
```

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Setup.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Setup.pm:152]: btw, I was also wondering if one week is a good default or if we should even go for a higher default.

File: etc/openqa/openqa.ini [hunk 3]
```diff
@@ -195,6 +195,8 @@ concurrent = 0
 # Extends the job result cleanup to ensure the partition results are stored on does not become too full
 # (still experimental, relies on df)
 #results_min_free_disk_space_percentage = 0
+##Duration to keep minion jobs in the database (Minion default is 2d)
+#expire_minion_jobs=604800
```
[the remark this answers is not in the document]
Comment (perlpunk) [reply etc/openqa/openqa.ini]: done

**Review by okurz — APPROVED** [review]

File: lib/OpenQA/Setup.pm (hunk 1 above)
Comment (perlpunk) [reply lib/OpenQA/Setup.pm]: done

File: lib/OpenQA/WebAPI.pm [hunk 4]
```diff
@@ -81,6 +81,9 @@ sub startup ($self) {
     # set cookie timeout to 48 hours (will be updated on each request)
     $self->app->sessions->default_expiration(48 * 60 * 60);
 
+    my $expire_minion = OpenQA::App->singleton->config->{misc_limits}->{expire_minion_jobs};
+    $self->app->minion->remove_after($expire_minion) if $expire_minion;
```
Comment (Martchus) [inline lib/OpenQA/WebAPI.pm:85]: I'm wondering whether this would be better set within the gru service. Likely @kraih knows.

Comment (kraih) [reply lib/OpenQA/WebAPI.pm:85]: Good point, it probably is cleaner to put the code into the `register` method in `OpenQA::Shared::Plugin::Gru`.

**Comment by kraih** [comment]
The word `expire` is not a good choice here, since it can be confused with the actual expiration feature of Minion (which can be used to delete jobs that have not been processed yet after a certain time).

**Comment by perlpunk** [comment]
> The word `expire` is not a good choice here, [...]

Do you have a better suggestion for a word?

File: lib/OpenQA/Setup.pm [hunk 5]
```diff
@@ -148,6 +149,7 @@ sub read_config ($app) {
             screenshot_cleanup_batch_size => OpenQA::Task::Job::Limit::DEFAULT_SCREENSHOTS_PER_BATCH,
             screenshot_cleanup_batches_per_minion_job => OpenQA::Task::Job::Limit::DEFAULT_BATCHES_PER_MINION_JOB,
             results_min_free_disk_space_percentage => undef,
+            expire_minion_jobs => ONE_WEEK,
```
Comment (kalikiana) [inline lib/OpenQA/Setup.pm:152]: For consistency this should start with a noun.
```suggestion
            minion_job_max_age => ONE_WEEK,
```

Comment (perlpunk) [reply lib/OpenQA/Setup.pm]: done

File: lib/OpenQA/WebAPI.pm (hunk 4 above)
Comment (perlpunk) [reply lib/OpenQA/WebAPI.pm]: done

**Review by kraih — APPROVED** [review]

**Pull request merged**